js/pages/cohort-definitions/components/checks/conceptset-warnings.js

```diff
@@ -11,7 +11,10 @@ define(['knockout', 'text!./conceptset-warnings.html',
     function conceptSetWarnings(params){
       var self = this;
       self.model = params.model;
-      self.cohortDefinitionId = self.model.currentCohortDefinition().id || ko.observable(-1);
+      self.cohortDefinitionId = ko.observable(self.model.currentCohortDefinition().id()) || ko.observable(-1);
+      self.model.cohortDefinitionSubscriptions.push(
+          self.model.currentCohortDefinition.subscribe((definition) => self.cohortDefinitionId(definition ? definition.id() : -1))
```

Comment:
the way this looks:  subscribe to model.currentCohortDefinition, on change update self.cohortDefinitionId either the new definitionId or -1.

Couldn't this be handled by:
`
self.cohortDefinitionId = ko.pureComputed(() => self.model.currentCohortDefinition() && self.model.currentCohortDefinition().id() || -1)
`

`